x-pack/plugins/alerting/server/monitoring/register_cluster_collector.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import stats from 'stats-lite';
+import { MonitoringCollectionSetup } from '../../../monitoring_collection/server';
+import { CoreSetup } from '../../../../../src/core/server';
+import { AlertingPluginsStart } from '../plugin';
+import { ClusterRulesMetric } from './types';
+
+export function registerClusterCollector({
+  monitoringCollection,
+  core,
+}: {
+  monitoringCollection: MonitoringCollectionSetup;
+  core: CoreSetup<AlertingPluginsStart, unknown>;
+}) {
+  monitoringCollection.registerMetric({
+    type: 'cluster_rules',
+    schema: {
+      overdue: {
+        count: {
+          type: 'long',
+        },
+        duration: {
+          p50: {
+            type: 'long',
+          },
+          p99: {
+            type: 'long',
+          },
+        },
+      },
+    },
+    fetch: async () => {
+      const services = await core.getStartServices();
+      const now = +new Date();
+      const { docs: overdueTasks } = await services[1].taskManager.fetch({
+        query: {
+          bool: {
+            must: [
+              {
+                term: {
+                  'task.scope': {
+                    value: 'alerting',
+                  },
+                },
+              },
+              {
+                bool: {
```

Comment:
same comment as above about reusing the query

Reply:
On it!